examples/DictUpdateExamples.ipynb

```diff
@@ -0,0 +1,130 @@
+    "    #note that when we add these traitlets, they will automatically be made available to the react component\n",
+    "    simple_dict = Dict({'foo':5, 'bar':8, 'baz':10}).tag(sync=True)\n",
+    "    count = Int(0).tag(sync=True)\n",
+    "\n",
+    "    @observe('count')\n",
+    "    def _observe_count(self, change):\n",
+    "        new_val = self.count\n",
+    "        sd = self.simple_dict.copy()\n",
+    "        sd['foo'] += 1\n",
+    "        self.simple_dict = sd\n",
+    "\n",
+    "    _esm = \"\"\"\n",
+    "    import confetti from \"canvas-confetti\";\n",
+    "    import * as React from \"react\";\n",
+    "    \n",
+    "    export const DictView = ({fullDict}) => {\n",
+    "        return (<table>\n",
+    "                  <tr><th>foo</th><th>bar</th><th>baz</th></tr>\n",
+    "                  <tr><td>{fullDict['foo']}</td><td>{fullDict['bar']}</td><td>{fullDict['baz']}</td></tr>\n",
+    "                </table>);\n",
+    "    }\n",
+    "\n",
+    "    export default function({on_count, debug, count, simple_dict}) {\n",
+    "        return <div><button onClick={() => confetti() && on_count(count + 1)}>\n",
+    "            {count} times confetti\n",
+    "        </button>\n",
+    "        <DictView fullDict={simple_dict} />\n",
+    "    </div>\n",
+    "    };\"\"\"\n",
+    "w = UpdateDictFromPythonWidget()\n",
+    "w"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "88858dd6-b2dd-4ee9-a22f-3b4a6e698212",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class UpdateDictFromReactWidget(ipyreact.ReactWidget):\n",
+    "    #note that when we add these traitlets, they will automatically be made available to the react component\n",
+    "    simple_dict = Dict({'foo':5, 'bar':8, 'baz':10}).tag(sync=True)\n",
+    "    count = Int(0).tag(sync=True)\n",
+    "\n",
+    "    @observe('count')\n",
+    "    def _observe_count(self, change):\n",
+    "        new_val = self.count\n",
+    "        sd = self.simple_dict.copy()\n",
+    "        sd['foo'] += 1\n",
+    "        self.simple_dict = sd\n",
+    "\n",
+    "    @observe('simple_dict')\n",
+    "    def _observe_simple_dict(self, change):\n",
+    "        print(self.simple_dict)\n",
+    "    _esm = \"\"\"\n",
+    "    import confetti from \"canvas-confetti\";\n",
+    "    import * as React from \"react\";\n",
+    "    \n",
+    "    export const DictView = ({fullDict}) => {\n",
+    "        return (<table>\n",
+    "                  <tr><th>foo</th><th>bar</th><th>baz</th></tr>\n",
+    "                  <tr><td>{fullDict['foo']}</td><td>{fullDict['bar']}</td><td>{fullDict['baz']}</td></tr>\n",
+    "                </table>);\n",
+    "    }\n",
+    "\n",
+    "    export const DictUpdate = ({fullDict, set_fullDict}) => {\n",
+    "        return (<button onClick={() => {\n",
+    "        console.log('dict update')\n",
+    "                                         const newVal = {...fullDict};\n",
```

Comment:
Also could use a comment here that we do a (shallow) copy so we don't mutate the original object, otherwise changes detection would go unnoticed.